### PR TITLE
[cmake] Do not operate on install directory at build time

### PR DESCRIPTION
**Pull request by eguiraud**
Solves the problem reported at https://github.com/lobis/root-docker/pull/1 .

### CODE REVIEW

File: CMakeLists.txt [hunk 1]
```diff
@@ -514,17 +514,17 @@ endif()
...
                      COMMAND ${CMAKE_COMMAND} -E remove -f modules.idx modules.timestamp
-                     COMMAND ${MODULES_ROOT_INCPATH} ${ld_library_path}=${CMAKE_LIBRARY_OUTPUT_DIRECTORY}:$ENV{${ld_library_path}}
+                     COMMAND ${ld_library_path}=${library_output_dir}:$ENV{${ld_library_path}}
                              ROOTIGNOREPREFIX=1 ROOT_HIST=0
                              $<TARGET_FILE:root.exe> -l -q -b
-                     WORKING_DIRECTORY ${CMAKE_INSTALL_LIBDIR}
+                     WORKING_DIRECTORY ${library_output_dir}
                      DEPENDS $<TARGET_FILE:root.exe> Cling Hist Tree Gpad Graf HistPainter move_artifacts ${modules_idx_deps})
-  add_custom_target(modules_idx ALL DEPENDS ${CMAKE_INSTALL_LIBDIR}/modules.idx)
+                  add_custom_target(modules_idx ALL DEPENDS ${library_output_dir}/modules.idx)
```
[the remark this answers is not in the document]
Comment (eguiraud) [reply CMakeLists.txt:525]: thanks! should it have 2 whitespaces in front? 
```suggestion
  add_custom_target(modules_idx ALL DEPENDS ${library_output_dir}/modules.idx)
```

[the remark this answers is not in the document]
Comment (bellenot) [reply CMakeLists.txt:525]: Yes

[the remark this answers is not in the document]
Comment (bellenot) [reply CMakeLists.txt:525]: > thanks! should it have 2 whitespaces in front?

yes

File: CMakeLists.txt [hunk 2]
```diff
@@ -514,17 +514,17 @@ endif()
...
                              ROOTIGNOREPREFIX=1 ROOT_HIST=0
                              $<TARGET_FILE:root.exe> -l -q -b
-                     WORKING_DIRECTORY ${CMAKE_INSTALL_LIBDIR}
+                     WORKING_DIRECTORY ${library_output_dir}
                      DEPENDS $<TARGET_FILE:root.exe> Cling Hist Tree Gpad Graf HistPainter move_artifacts ${modules_idx_deps})
-  add_custom_target(modules_idx ALL DEPENDS ${CMAKE_INSTALL_LIBDIR}/modules.idx)
+  add_custom_target(modules_idx ALL DEPENDS ${library_output_dir}/modules.idx)
   add_dependencies(modules_idx ${modules_idx_deps})
-  set_directory_properties(PROPERTIES ADDITIONAL_CLEAN_FILES "${CMAKE_INSTALL_LIBDIR}/modules.timestamp")
+  set_directory_properties(PROPERTIES ADDITIONAL_CLEAN_FILES "${library_output_dir}/modules.timestamp")
```
Comment (vgvassilev) [inline CMakeLists.txt:527]: Do we still install these files? I think we should.

Comment (eguiraud) [reply CMakeLists.txt:527]: `PROPERTIES ADDITIONAL_CLEAN_FILES` does not change what files are installed.

in general, this PR changes what directories are involved in a `add_custom_command` which is executed at build time, so it should not change what is installed or where.

And I can see a file `modules.idx` being installed besides our `libROOT*.so` ilbraries in the docker image that I built with this patch.

So I think it's all good on the installation front?

Comment (vgvassilev) [reply CMakeLists.txt:527]: Yep, thanks. Go ahead!

**Pull request merged**